### PR TITLE
Design docs

### PR DESCRIPTION
**Pull request by robhoes**
Moving the final bits (well, nearly...) from xapi-project.github.io from Jekyll to Hugo.

### CODE REVIEW

**Review by psafont — APPROVED** [review]

File: doc/content/design/RDP.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: RDP control
+layout: default
+design_doc: true
+revision: 2
+status: released (XenServer 6.5 SP1)
+design_review: 12
```
Comment (gangj) [inline doc/content/design/RDP.md:7]: I see this field in the original [page](https://xapi-project.github.io/features/RDP.html) is a link to: http://github.com/xapi-project/xapi-project.github.io/issues/12, do we need to keep it?

File: doc/content/design/_index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
++++
+title = "Design Documents"
+menuTitle = "Designs"
++++
+
+{{< design_docs_list >}}
```
Comment (gangj) [inline doc/content/design/_index.md:6]: Sorry, will the `design_docs_list` be rendered as the list of docs under the folder? But I don't see it is rendered by the browser.

File: doc/layouts/partials/content-header.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{{ if eq $.Page.Params.design_doc true }}
```
Comment (gangj) [inline doc/layouts/partials/content-header.html:1]: May I know how will this page be rendered? Is it a page provided similar as homepage by github with Hugo, in that way, a page will be available under: https://xen-api.github.io/

File: doc/content/design/_index.md (hunk 2 above)
Comment (gangj) [reply doc/content/design/_index.md:6]: Oh, I see it: [doc/layouts/shortcodes/design_docs_list.html](https://github.com/xapi-project/xen-api/pull/5664/files#diff-f9479e661726cb84c3528443dd06c989535f7f004fc0605a990551a58eb1933c)

**Review by gangj — APPROVED** [review]

File: doc/content/design/_index.md (hunk 2 above)
Comment (robhoes) [reply doc/content/design/_index.md:6]: Yes, the `{{< ... >}}` bit refers to a "shortcode" in Hugo, which is something that can be included in a markdown doc and is automatically rendered as HTML.

File: doc/layouts/partials/content-header.html (hunk 3 above)
Comment (robhoes) [reply doc/layouts/partials/content-header.html:1]: It's a table with metadata about the design like on the old pages:

E.g.: 
Screenshot 2024-06-04 at 13 58 02

File: doc/content/design/RDP.md (hunk 1 above)
Comment (robhoes) [reply doc/content/design/RDP.md:7]: Yes, I'll add those links, just for historical reasons.

File: doc/content/design/RDP.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+Note that the current behaviour depends on some global options: "Enable Remote Desktop console scanning" and "Automatically switch to the Remote Desktop console when it becomes available".
+
+1.    When tools are not installed:
+  *       As of XenCenter 6.5, the RDP button is absent.
+2.    When tools are installed but RDP is not switched on in the guest:
+  1.      If "Enable Remote Desktop console scanning" is on:
+    *         The RDP button is present but greyed out. (It seems to sometimes read "Switch to Remote Desktop" and sometimes read "Looking for guest console...": I haven't yet worked out the difference).
+    *         We scan the RDP port to detect when RDP is turned on
+  2.      If "Enable Remote Desktop console scanning" is off:
+    *         The RDP button is enabled and reads "Switch to Remote Desktop"
```
[the remark this answers is not in the document]
Comment (robhoes) [reply doc/content/design/RDP.md]: Even the old one was bad! I've added a comment to fix it.

**Pull request merged**